Sources/Apodini/Components/Component.swift

```diff
@@ -7,37 +7,84 @@
 
 import NIO
 @_implementationOnly import Runtime
+@_implementationOnly import AssociatedTypeRequirementsVisitor
 
 
 /// A `Component` is the central building block of  Apodini. Each component handles a specific functionality of the Apodini web service.
 ///
-/// A `Component` either has a `handle` function that is called when a request reaches the `Component` or consists of different other components as described by the `content` property.
+/// A `Component`  consists of different other components as described by the `content` property.
 public protocol Component {
     /// The type of `Component` this `Component` is made out of if the component is a composition of multiple subcomponents.
-    associatedtype Content: Component = Never
-    /// The type that is returned from the `handle` method when the component handles a request. The return type of the `handle` method is encoded into the response send out to the client.
-    associatedtype Response: Encodable = Never
-    
+    associatedtype Content: Component
     
     /// Different other `Component`s that are composed to describe the functionality of the`Component`
-    @ComponentBuilder var content: Self.Content { get }
+    @ComponentBuilder
+    var content: Content { get }
+}
+
+
+/// A `Handler` is a `Component` which defines an endpoint and can handle requests.
+public protocol Handler: Component {
+    /// The type that is returned from the `handle()` method when the component handles a request. The return type of the `handle` method is encoded into the response send out to the client.
+    associatedtype Response: Encodable
     
+    /// A function that is called when a request reaches the `Handler`
+    func handle() -> Response
+}
+
+
+extension Handler {
+    public var content: some Component { EmptyComponent() }
+}
+
+
+/// A `Handler` which can be uniquely identified
+public protocol IdentifiableHandler: Handler {
+    /// The type of this handler's identifier
+    associatedtype EndpointIdentifier: AnyEndpointIdentifier
     
-    /// A function that is called when a request reaches the `Component`
-    func handle() -> Self.Response
+    /// This handler's identifier
+    var endpointId: EndpointIdentifier { get }
```

Comment:
I would suggest to name this property handlerId and the corresponding types to be renamed to `HandlerIdentifier` as we do not use the terminology endpoint in the public facing API and only in the internal APIs and the semantic model.
```suggestion
    var handlerId: HandlerIdentifier { get }
```
The only element that would speak against this would be that the identifier will be eventually be used to identify an endpoint but I would be fine with using the term `HandlerIdentifier` in the `Endpoint` struct or using a type alias in there to name it `EndpointIdentifier` in the context where we start referring to it as an endpoint. What do you think @lukaskollmer ?

Reply:
Yeah the reason I chose `EndpointIdentifier` over `HandlerIdentifier` was precisely because at the end of the day we think of the web service's exported interface as a set of endpoints, not handlers.
Not sure about the typealias, since we'd just introduce a second name for the same thing (in the context where we're referring to it as an endpoint, the `HandlerIdentifier` type would also still be available), and it might cause confusion as to why there's both `HandlerIdentifier` and `EndpointIdentifier`.

I'm fine with it either way, but I'll rename it so that the terminology matches.